smoke/features/steps/steps.py

```diff
@@ -153,4 +151,65 @@ def jenkinsMasterPodStatus(context):
 
 @given(u'The jenkins pod is up and runnning')
 def checkJenkins(context):
-    jenkinsMasterPodStatus(context)
\ No newline at end of file
+    jenkinsMasterPodStatus(context)
+@then(u'we configure custom agents as Kubernetes pod template by creating configmap using "smoke/samples/java-builder-cm.yaml" and "smoke/samples/nodejs-builder-cm.yaml"')
+def configure_pod_templates(context):
+    print("Initiazing java pod template")
+    res = oc.oc_create_from_yaml(java_builder)
+    if(res == None):
+        print("Error while creating java-builder pod template")
+        raise AssertionError
+    else:
+        print("Initiazing nodejs pod template")
+        nodejs_res = oc.oc_create_from_yaml(nodejs_builder)
+        if(nodejs_res == None):
+            print("Error while creating nodejs-builder pod template")
+            raise AssertionError
+@then(u'we check configmap "jenkins-agent-java-builder" and "jenkins-agent-nodejs" should be created')
+def verify_configmap(context):
+    current_project = context.current_project
+    if not 'jenkins-agent-java-builder' in oc.search_resource_in_namespace('configmap','jenkins-agent-java-builder',current_project):
+        raise AssertionError
+    elif not 'jenkins-agent-nodejs' in oc.search_resource_in_namespace('configmap','jenkins-agent-nodejs',current_project):
+        raise AssertionError
+@when(u'the user creates a new build refering to "https://github.com/akram/pipes.git\#container-nodes"')
+def trigger_new_build(context):
+    res = oc.new_build(podtemplate_build_ref)
+    time.sleep(30)
+    if res == None:
+        raise AssertionError
+@then(u'buildconfig.build.openshift.io "pipes" should be created')
+def search_buildconfig(context):
+    current_project = context.current_project
+    item = oc.search_resource_in_namespace('buildconfig', 'pipes', current_project)
+    if item is None:
+        raise AssertionError
+    else:
+        print(f'buildconfig {item} found')
+    
+@then(u'we check for "java-builder" agent node and "nodejs-builder" agent node are created')
+def check_pods(context):
+    current_project = context.current_project
+    time.sleep(75)
```

Comment:
that's a long sleep time, we need something more flexible and able to fail faster in case of an error.